src/web/pages/portlists/PortListComponent.jsx

```diff
@@ -7,220 +7,223 @@ import _ from 'gmp/locale';
+      const response = await gmp.portlist.import(data);
+      onImported(response);
+      closeImportDialog();
+    } catch (error) {
+      onImportError(error);
+    }
+  };
 
-    this.handleInteraction();
+  const handleSavePortList = async data => {
+    handleInteraction();
 
-    let promises = created_port_ranges_copy.map(range => {
-      const saveData = {
+    const createdPromises = createdPortRanges.map(async range => {
+      // save temporary port ranges in the backend
+      const id = await handleSavePortRange({
         ...range,
         port_range_start: parseInt(range.start),
         port_range_end: parseInt(range.end),
         port_type: range.protocol_type,
-      };
-      return this.handleSavePortRange(saveData).then(id => {
-        range.isTmp = false;
-        range.id = id;
-        this.created_port_ranges = this.created_port_ranges.filter(
-          prange => prange !== range,
-        );
       });
+      range.isTmp = false;
+      range.id = id;
+      // the range has been saved in the backend
+      // if something fails the state contains the still to be saved ranges
+      setCreatedPortRanges(createdPortRanges =>
+        createdPortRanges.filter(pRange => pRange !== range),
+      );
     });
-    const deleted_port_ranges_copy = [...this.deleted_port_ranges];
-    promises = [
-      ...promises,
-      ...deleted_port_ranges_copy.map(range =>
-        this.handleDeletePortRange(range).then(
-          (this.deleted_port_ranges = this.deleted_port_ranges.filter(
-            prange => prange !== range,
-          )),
-        ),
-      ),
-    ];
-    return Promise.all(promises)
-      .then(() => save(data))
-      .then(() => this.closePortListDialog())
-      .catch(error => {
-        if (data?.id && isDefined(this.props.onSaveError)) {
-          return this.props.onSaveError(error);
-        } else if (!data?.id && isDefined(this.props.onCreateError)) {
-          return this.props.onCreateError(error);
-        }
-        return Promise.reject(error);
-      });
-  }
+    const deletedPromises = deletedPortRanges.map(async range => {
+      await handleDeletePortRange(range);
+      // the range has been deleted from the backend
+      // if something fails the state contains the still to be deleted ranges
+      setDeletedPortRanges(deletedPortRanges =>
+        deletedPortRanges.filter(pRange => pRange !== range),
+      );
+    });
+
+    const promises = [...createdPromises, ...deletedPromises];
+    try {
+      await Promise.all(promises);
+    } catch (error) {
+      if (isDefined(data?.id) && isDefined(onSaveError)) {
+        return onSaveError(error);
+      } else if (!isDefined(data?.id) && isDefined(onCreateError)) {
+        return onCreateError(error);
+      }
+      throw error;
+    }
+    await handleSave(data);
+    closePortListDialog();
+  };
 
-  handleTmpAddPortRange(values) {
-    const {port_ranges} = this.state;
+  const handleTmpAddPortRange = async values => {
```

Comment:
I would say we do not need it if there is no `await`.

```suggestion
  const handleTmpAddPortRange = values => {
```